internal/services/azurestackhci/stack_hci_marketplace_gallery_image_resource.go

```diff
@@ -0,0 +1,346 @@
+	Name              string                                      `tfschema:"name"`
+	ResourceGroupName string                                      `tfschema:"resource_group_name"`
+	Location          string                                      `tfschema:"location"`
+	CustomLocationId  string                                      `tfschema:"custom_location_id"`
+	HypervGeneration  string                                      `tfschema:"hyperv_generation"`
+	Identifier        []StackHCIMarketplaceGalleryImageIdentifier `tfschema:"identifier"`
+	OsType            string                                      `tfschema:"os_type"`
+	Version           string                                      `tfschema:"version"`
+	StoragePathId     string                                      `tfschema:"storage_path_id"`
+	Tags              map[string]interface{}                      `tfschema:"tags"`
+}
+
+type StackHCIMarketplaceGalleryImageIdentifier struct {
+	Offer     string `tfschema:"offer"`
+	Publisher string `tfschema:"publisher"`
+	Sku       string `tfschema:"sku"`
+}
+
+func (StackHCIMarketplaceGalleryImageResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9][\-\.\_a-zA-Z0-9]{0,78}[a-zA-Z0-9]$`),
+				"name must be between 2 and 80 characters and can only contain alphanumberic characters, hyphen, dot and underline",
+			),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"custom_location_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: customlocations.ValidateCustomLocationID,
+		},
+
+		"hyperv_generation": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(marketplacegalleryimages.HyperVGenerationVOne),
+				string(marketplacegalleryimages.HyperVGenerationVTwo),
+			}, false),
+		},
+
+		"identifier": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"publisher": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"offer": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"sku": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"os_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
```

Comment:
please use `PossibleValuesFor` method if not exist `None` enum value

Reply:
fixed